tests/src/test_set.cpp

```diff
@@ -85,6 +99,19 @@ SCENARIO( "a Set can be interacted with by consuming classes even when const", "
 				CHECK( secondPtr == firstPtr );
 			}
 		}
+
+		WHEN( "find(key) is called" ) {
+			const auto cIt = s.find(key);
+			THEN( "the Set does not increase in size" ) {
+				CHECK( s.size() == 1 );
+			}
+			THEN( "the iterator does point to the begin position" ) {
+				CHECK( cIt == s.begin() );
+			}			
+			THEN( "the iterator doesn't point to the end position" ) {
+				CHECK( cIt != s.end() );
+			}			
```

Comment:
This should be a REQUIRE, immediately after the call to find, and outside of a THEN.